bool initialized_
uint64 data